lib/ansible/module_utils/azure_rm_common.py

```diff
@@ -0,0 +1,635 @@
+        the object.
+
+        :param tags: object metadata tags
+        :return: bool, dict of tags
+        '''
+        if not self.module.params.get('tags'):
+            # purge all tags
+            return True, dict()
+        new_tags = copy.copy(tags)
+        changed = False
+        for key in tags:
+            if not self.module.params['tags'].get(key):
+                # key not found in user provided parameters
+                new_tags.pop(key)
+                changed = True
+        if changed:
+            self.log('CHANGED: purged tags')
+        return changed, new_tags
+
+    def _tag_update(self, tags):
+        '''
+        Update metadata tags with values in user provided tags parameter. Returns
+        tuple with bool indicating something changed and dict of new tags to be
+        assigned to the object.
+
+        :param tags: object metadata tags
+        :return: bool, dict of tags
+        '''
+        if isinstance(tags, dict):
+            new_tags = copy.copy(tags)
+        else:
+            new_tags = dict()
+        changed = False
+        if self.module.params.get('tags'):
+            for key, value in self.module.params['tags'].iteritems():
+                if not (new_tags.get(key) and new_tags[key] == value):
+                    changed = True
+                    new_tags[key] = value
+        if changed:
+            self.log('CHANGED: updated tags')
+        return changed, new_tags
+
+    def update_tags(self, tags):
+        '''
+        Call from the module to update metadata tags. Returns tuple
+        with bool indicating if there was a change and dict of new
+        tags to assign to the object.
+
+        :param tags: metadata tags from the object
+        :return: bool, dict
+        '''
+        changed = False
+        updated, new_tags = self._tag_update(tags)
+        if updated:
+            changed = True
+
+        if self.module.params['purge_tags']:
+            purged, new_tags = self._tag_purge(new_tags)
+            if purged:
+                changed = True
+        return changed, new_tags
+
+    def exec_module(self):
+        '''
+        Execute module implementation.
+
+        :return: None
+        '''
+        res = self.exec_module_impl(**self.module.params)
+        self.module.exit_json(**res)
+
+    def get_resource_group(self, resource_group):
+        '''
+        Fetch a resource group.
+
+        :param resource_group: name of a resource group
+        :return: resource group object
+        '''
+        try:
+            return self.rm_client.resource_groups.get(resource_group)
+        except CloudError:
+            self.fail("Parameter error: resource group {0} not found".format(resource_group))
+        except Exception, exc:
+            self.fail("Error retrieving resource group {0} - {1}".format(resource_group, str(exc)))
+
+    def _get_profile(self, profile="default"):
+        path = expanduser("~")
+        path += "/.azure/credentials"

```

Comment:
trivial but you can just do:
``` python
path = expanduser("~/.azure/credentials")
```
so that you only allocate a string once instead of three times.  (+= creates a new string)